### PR TITLE
Adds lighting height control (space color consistency)

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

Adds support for modifying a light's "height"
You can think of this as the distance it is from the ground below it (Really it's the distance to the corners around it + 0.5 but yaknow) We use it to keep wall lights from looking weird, but well, not everything is a wall light

Floors tend to not be, and space in particular does not want to be treated as such.
In fact, it wants a NEGATIVE height, so it acts as if it in on top of all of its corners. This allows us to ensure that the starlight from space and the starlight from starlight overlays always have the same intensity and color, preventing weird lines from where the two intersect, or starlight feeling not very present in cases with only one turf

I've also bumped starlight's intensity form 0.75 to 1, this should help with the lines thing discussed above.

## Why It's Good For The Game

image

## Changelog
:cl:
add: Starlight should be a bit more intense, and flow better onto non space tiles
/:cl:

### CODE REVIEW

**Comment by MTandi** [comment]
Looks like this when `area_flags = AREA_USES_STARLIGHT`

image

**Comment by LemonInTheDark** [comment]
Try with 78877 (once I merge this to latest)
area starlight flags are a spook

**Comment by LemonInTheDark** [comment]
image

File: code/__DEFINES/dcs/signals/signals_atom/signals_atom_lighting.dm [hunk 1]
```diff
@@ -31,6 +31,10 @@
 #define COMSIG_ATOM_SET_LIGHT_ON "atom_set_light_on"
 ///Called right after the atom changes the value of light_on to a different one, from base of [/atom/proc/set_light_on]: (old_value)
 #define COMSIG_ATOM_UPDATE_LIGHT_ON "atom_update_light_on"
+///Called right before the atom changes the value of leight_height to a different one, from base [atom/proc/set_leight_height]: (new_value)
+#define COMSIG_ATOM_SET_LIGHT_HEIGHT "atom_set_leight_height"
+///Called right after the atom changes the value of leight_height to a different one, from base of [/atom/proc/set_leight_height]: (old_value)
+#define COMSIG_ATOM_UPDATE_LIGHT_HEIGHT "atom_update_leight_height"
```
Comment (san7890) [inline code/__DEFINES/dcs/signals/signals_atom/signals_atom_lighting.dm:37]:
```suggestion
///Called right before the atom changes the value of light_height to a different one, from base [atom/proc/set_light_height]: (new_value)
#define COMSIG_ATOM_SET_LIGHT_HEIGHT "atom_set_light_height"
///Called right after the atom changes the value of light_height to a different one, from base of [/atom/proc/set_light_height]: (old_value)
#define COMSIG_ATOM_UPDATE_LIGHT_HEIGHT "atom_update_light_height"
```

File: code/__DEFINES/lighting.dm [hunk 2]
```diff
@@ -28,6 +28,10 @@
 #define LIGHTING_FALLOFF 1
 /// use lambertian shading for light sources
 #define LIGHTING_LAMBERTIAN 0
+/// light UNDER the floor. primarially used for starlight, shouldn't fuck with this
```
Comment (san7890) [inline code/__DEFINES/lighting.dm:31]:
```suggestion
/// light UNDER the floor. primarily used for starlight, shouldn't fuck with this
```

**Review by san7890 — CHANGES_REQUESTED** [review]
spelling nits

**Pull request merged**